File: src/main/java/no/nav/syfo/config/FastlegeInformasjonConfig.java
package no.nav.syfo.config;

import no.nav.syfo.consumer.util.ws.LogErrorHandler;
import no.nav.syfo.consumer.util.ws.WsClient;
import no.nhn.schemas.reg.flr.IFlrReadOperations;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.*;

import static java.util.Collections.singletonList;
import static no.nav.syfo.consumer.util.ws.STSClientConfig.configureRequestSamlToken;

@Configuration
public class FastlegeInformasjonConfig {


    @Bean
    @Primary
    @ConditionalOnProperty(value="mockEksternHelse", havingValue = "false", matchIfMissing = true)
    public IFlrReadOperations fastlegeSoapClient(
            @Value("${ekstern.helse.fastlegeinformasjon.url}") String serviceUrl) {

        IFlrReadOperations port = new WsClient<IFlrReadOperations>().createPort(
                serviceUrl,
                IFlrReadOperations.class,
                singletonList((new LogErrorHandler())));
        configureRequestSamlToken(port);
        return port;

    }

}

File: src/main/kotlin/no/nav/syfo/consumer/ws/fastlege/FastlegeInformasjonConfig.kt
package no.nav.syfo.consumer.ws.fastlege

import no.nav.syfo.consumer.util.ws.LogErrorHandler
import no.nav.syfo.consumer.ws.util.*
import no.nhn.schemas.reg.flr.IFlrReadOperations
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.context.annotation.*

@Configuration
class FastlegeInformasjonConfig {
    @Bean
    @Primary
    @ConditionalOnProperty(value = ["mockEksternHelse"], havingValue = "false", matchIfMissing = true)
    fun fastlegeSoapClient(
        @Value("\${ekstern.helse.fastlegeinformasjon.url}") serviceUrl: String): IFlrReadOperations {
        val port = WsClient<IFlrReadOperations>().createPort(
            serviceUrl,
            IFlrReadOperations::class.java,
            listOf(LogErrorHandler()))
        STSClientConfig.configureRequestSamlToken(port)
        return port
    }
}
